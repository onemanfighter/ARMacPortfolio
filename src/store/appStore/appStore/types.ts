import { StateCreator } from 'zustand';
import { FinderStateSlice } from '../slice/Finder/types';
import { BinStateSlice } from '../slice';
import { TerminalStateSlice } from '../slice/Terminal';

export interface AppStoreState {
  Finder: FinderStateSlice;
  Bin: BinStateSlice;
  Terminal: TerminalStateSlice;
}

export type AppStoreSlice<T> = StateCreator<
  AppStoreState,
  [['zustand/immer', never]],
  [],
  T
>;
