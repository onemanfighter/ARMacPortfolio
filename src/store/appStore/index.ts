export * from './appStore';
export * from './slice';
export * from './selector';
