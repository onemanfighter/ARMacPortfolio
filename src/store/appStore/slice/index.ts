export * from './Finder';
export * from './Bin';
