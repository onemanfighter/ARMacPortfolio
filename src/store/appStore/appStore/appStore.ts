import { create } from 'zustand';

import { immer } from 'zustand/middleware/immer';
import { createJSONStorage, persist } from 'zustand/middleware';
import { AppStoreState } from './types';
import {
  createBinSlice,
  createChromeSlice,
  createFinderSlice,
  createGithubSlice,
  createTerminalSlice,
  createVsCodeSlice,
} from '../slice';
import { createSpotifySlice } from '../slice/Spotify';

export const appStore = create<AppStoreState>()(
  persist(
    immer((...api) => ({
      Finder: createFinderSlice(...api),
      Bin: createBinSlice(...api),
      Terminal: createTerminalSlice(...api),
      VsCode: createVsCodeSlice(...api),
      Chrome: createChromeSlice(...api),
      Spotify: createSpotifySlice(...api),
      Github: createGithubSlice(...api),
    })),
    {
      name: 'appStore',
      storage: createJSONStorage(() => localStorage),
    },
  ),
);
