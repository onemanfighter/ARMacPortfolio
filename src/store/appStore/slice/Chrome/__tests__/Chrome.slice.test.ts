import { renderHook } from '@testing-library/react-hooks';
import { appStore } from '../../../appStore';

describe('Chrome slice', () => {
  it('should return default chrome state', () => {
    renderHook(() => appStore());
    const { result } = renderHook(() => appStore());

    expect(result.current.Chrome.linkStack).toMatchSnapshot();
  });

  it('should add link to linkStack', () => {
    const { result } = renderHook(() => appStore());

    result.current.Chrome.addLinkStack('https://www.google.com/webhp?igu=1');

    expect(result.current.Chrome.linkStack).toMatchSnapshot();
  });

  it('should remove link from linkStack', () => {
    const { result } = renderHook(() => appStore());

    result.current.Chrome.addLinkStack('https://www.google.com/webhp?igu=1');
    result.current.Chrome.addLinkStack('https://www.google.com/webhp?igu=2');
    result.current.Chrome.removeLinkStack();

    expect(result.current.Chrome.linkStack).toMatchSnapshot();
  });

  it('should clear linkStack', () => {
    const { result } = renderHook(() => appStore());

    result.current.Chrome.addLinkStack('https://www.google.com/webhp?igu=1');
    result.current.Chrome.addLinkStack('https://www.google.com/webhp?igu=2');
    result.current.Chrome.clearLinkStack();

    expect(result.current.Chrome.linkStack).toMatchSnapshot();
  });
});
