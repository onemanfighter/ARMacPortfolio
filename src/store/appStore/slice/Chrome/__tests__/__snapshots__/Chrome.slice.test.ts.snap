// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chrome slice should return default chrome state 1`] = `
{
  "homepageLink": "http://www.google.com/webhp?igu=1",
}
`;
